lib/engine_http.js

```diff
@@ -207,12 +207,26 @@ HttpEngine.prototype.step = function step(requestSpec, ee, opts) {
         }
 
         function requestCallback(err, res, body) {
-          debug('request: %s', JSON.stringify({
-            uri: requestParams.uri,
-            method: requestParams.method,
-            headers: requestParams.headers,
-            json: requestParams.json
-          }, null, 2));
+          if (process.env.DEBUG) {
+            let requestInfo = {
+              uri: requestParams.uri,
+              method: requestParams.method,
+              headers: requestParams.headers
+            };
+            if (requestParams.json && typeof requestParams.json !== 'boolean') {
+              requestInfo.json = requestParams.json;
+            }
+
+            // If "json" is set to an object, it will be serialised and sent as body and the value of the "body" attribute will be ignored.
+            if (requestParams.body && typeof requestParams.json !== 'object') {
+              // Only show the beginning of long bodies
+              requestInfo.body = requestParams.body.substring(0, 256);
```

Comment:
Good call, perhaps both can be done. An alternative is to add a way to dump everything into a file for a proper inspection afterwards, and keep this for checking that things look OK as a test is running.